pybleau/app/plotting/heatmap_factory.py

```diff
@@ -29,13 +31,35 @@ def _plot_data_single_renderer(self, x_arr=None, y_arr=None, z_arr=None,
                                    **adtl_arrays):
         """ Build plot data when single renderer is present.
         """
-        data_map = {self.x_col_name: x_arr, self.y_col_name: y_arr}
-        data_map.update(adtl_arrays)
+        # Collect all labels and reset x_arr as an int list
+        if x_arr.dtype in CATEGORICAL_TYPES:
+            if not self.x_labels:
+                self.x_labels = Series(x_arr).unique().tolist()
+
+            x_arr = np.array([self.x_labels.index(val)
+                              for val in self.x_labels])
```

Comment:
`x_arr = np.array(range(len(self.x_labels)))` is the same? Or maybe there is something I am not seeing here. Just a personal preference, I guess.

Reply:
You were correct, done. Great catch!